frontend/src/_ui/FolderList/FolderList.jsx

```diff
@@ -16,20 +20,61 @@ function FolderList({
   isLoading = false,
   renderBadgeForItems = [],
   renderBadge = null,
+  overLayComponent,
+  darkMode,
+  toolTipText,
   ...restProps
 }) {
+  const [isHovered, setIsHovered] = useState(false);
+  const [isHoveredInside, setIsHoveredInside] = useState(false);
+  const [showGroupOptions, setShowGroupOptions] = useState(false);
+  const target = useRef(null);
+
+  const handleMouseEnter = () => {
+    setIsHovered(true);
+  };
+
+  const handleMouseLeave = () => {
+    setIsHovered(false);
+  };
+
+  const menuToggle = () => {
+    setShowGroupOptions(!showGroupOptions);
+  };
+  const handleMouseEnterInside = () => {
+    setIsHoveredInside(true);
+  };
+
+  const handleMouseLeaveInside = () => {
+    setIsHoveredInside(false);
+  };
+
   return (
     <>
       {!isLoading ? (
         <button
           {...restProps}
           className={`tj-list-item ${selectedItem && 'tj-list-item-selected'}  ${className} ${
```

Comment:
If we you want to use more conditional classnames. lets try this.  
```suggestion
import cx from 'classnames';
      className={cx(`tj-list-item ${className}`, {
            'tj-list-item-selected': selectedItem,
            'tj-list-item-disabled': disabled,
            'tj-list-item-option-opened': showGroupOptions,
          })}
```